asynciterator.ts

```diff
@@ -309,6 +309,35 @@ export class AsyncIterator<T> extends EventEmitter {
     return '';
   }
 
+  /**
+    Consume all remaining items of the iterator into an array that will be returned asynchronously.
+    @param {object} [options] Settings for array creation
+    @param {integer} [options.limit] The maximum number of items to place in the array.
+   */
+  toArray(options?: { limit?: number }): Promise<T[]> {
+    return new Promise<T[]>((resolve, reject) => {
+      // Collect items in an array
+      const items: T[] = [];
+
+      // Create a named listener so we can easily remove it later
+      const dataListener = (item: T) => {
+        // Resolve earlier if a limit was set and we have reached that limit
+        // @ts-ignore
+        if (items.length >= options?.limit) {
+          this.removeListener('data', dataListener);
+          resolve(items);
+        }
+        else {
+          items.push(item);
+        }
```

Comment:
The conditional pushing is not going to work; we will be reading 1 too many elements time. So we will need the old logic and the zero check on top.